scripts/src/main/java/com/mathewsachin/fategrandautomata/scripts/modules/ServantTracker.kt

```diff
@@ -186,36 +186,34 @@ class ServantTracker @Inject constructor(
         }
 
         val cardsRemaining = CommandCard.Face.list.toMutableSet()
-        val result = mutableMapOf<TeamSlot, MutableSet<CommandCard.Face>>()
+        val result = mutableMapOf<TeamSlot, Set<CommandCard.Face>>()
 
         supportSlot?.let { supportSlot ->
             if (supportSlot in deployed.values) {
                 val matched = cardsRemaining.filter { card ->
                     images[Images.Support] in locations.attack.supportCheckRegion(card)
-                }.toMutableSet()
+                }.toSet()
 
                 cardsRemaining -= matched
-
                 result[supportSlot] = matched
             }
         }
 
         val ownedServants = faceCardImages
             .filterKeys { it != supportSlot && it in deployed.values }
-        for (card in cardsRemaining) {
+        cardsRemaining.groupBy { card ->
             // find the best matching Servant which isn't the support
```

Comment:
It would be easier to read if groupBy started from next line, the operators would be indented on the same level then. Same for mapValues down below.
But this is just me nitpicking

Reply:
```kt
cardsRemaining
    .groupBy { card ->
        // find the best matching Servant which isn't the support
        ownedServants
            .mapValues { (_, image) ->
                locations.attack.servantMatchRegion(card)
                    .find(image, 0.5)?.score ?: 0.0
            }
            .filterValues { it > 0.0 }
            .maxByOrNull { it.value }
            ?.key
    }
    .filterKeys { it != null }
    .entries
    .associateTo(result) { (key, value) -> key!! to value.toSet() }
```

Reply:
Done, didn't know what caused the extra indents at the bottom.

Reply:
The IDE only indents correctly if the operators are on new line for some reason.

Can you do the same for mapValues too like in my previous comments? It looks weird right now 😅

Reply:
Done